[Automation] java run fails for org.apache.hadoop:hadoop-auth:2.8.2

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25918434823/job/76181855729
Last 300 lines of the log:
```console
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE

> Task :compileTestJavaNote: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.


> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null

3 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date


> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.hadoop:hadoop-auth:2.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.8.2 at 2026-05-15T13:29:07Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 2.8.2 failed with exit code 1.
SOFT-FAIL[native-image run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean test -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.8.2 at 2026-05-15T13:29:07Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean compileTestJava -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 749ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 851ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.8.2 at 2026-05-15T13:29:10Z with exit code 0 after 3s
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage java run
========================================================================================
Starting stage 'java run' for 2.8.2 at 2026-05-15T13:29:10Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 762ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processTestResources NO-SOURCE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

3 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.8.2 at 2026-05-15T13:29:14Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.8.2 failed with exit code 1.
FAILED[java run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25918434823/job/76181855721
Last 300 lines of the log:
```console
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)

3 tests completed, 1 failed
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).


BUILD FAILED in 2s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.hadoop:hadoop-auth:2.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.8.2 at 2026-05-15T13:25:03Z with exit code 1 after 5s
❌ ERROR: 'native-image run' for 2.8.2 failed with exit code 1.
SOFT-FAIL[native-image run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean test -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.8.2 at 2026-05-15T13:25:03Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean compileTestJava -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 803ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 856ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.8.2 at 2026-05-15T13:25:05Z with exit code 0 after 2s
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage java run
========================================================================================
Starting stage 'java run' for 2.8.2 at 2026-05-15T13:25:05Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 772ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)

3 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.8.2 at 2026-05-15T13:25:09Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.8.2 failed with exit code 1.
FAILED[java run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25918434823/job/76181855706
Last 300 lines of the log:
```console
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

3 tests completed, 1 failed

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.hadoop:hadoop-auth:2.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.8.2 at 2026-05-15T13:32:12Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 2.8.2 failed with exit code 1.
SOFT-FAIL[native-image run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.8.2" ./gradlew clean test -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.8.2 at 2026-05-15T13:32:12Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.8.2" ./gradlew clean compileTestJava -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 814ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.

Note: Recompile with -Xlint:deprecation for details.
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 894ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.8.2 at 2026-05-15T13:32:15Z with exit code 0 after 3s
========================================================================================
 org.apache.hadoop:hadoop-auth:2.2.0:2.8.2 stage java run
========================================================================================
Starting stage 'java run' for 2.8.2 at 2026-05-15T13:32:15Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 776ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE

> Task :compileTestJava
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :testClasses

> Task :test FAILED

AuthenticatedURLTest > opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "hadoop.auth="delegation-token""
     but was: null
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//org_apache_hadoop.hadoop_auth.AuthenticatedURLTest.opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet(AuthenticatedURLTest.java:31)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

3 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.hadoop/hadoop-auth/2.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.8.2 at 2026-05-15T13:32:19Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.8.2 failed with exit code 1.
FAILED[java run][2.8.2][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.8.2" ./gradlew clean javaTest -Pcoordinates="org.apache.hadoop:hadoop-auth:2.2.0"]
```

--- a/metadata/org.apache.hadoop/hadoop-auth/2.8.2/reachability-metadata.json
+++ b/metadata/org.apache.hadoop/hadoop-auth/2.8.2/reachability-metadata.json
@@ -1,0 +1,71 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "type": "java.lang.Thread",
+      "methods": [
+        {
+          "name": "getContextClassLoader",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "type": "org.apache.hadoop.security.authentication.client.KerberosAuthenticator",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.util.KerberosUtil"
+      },
+      "type": "sun.security.jgss.GSSUtil",
+      "fields": [
+        {
+          "name": "GSS_KRB5_MECH_OID"
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "glob": "META-INF/services/com.sun.net.httpserver.spi.HttpServerProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "glob": "log4j.properties"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.hadoop.security.authentication.client.AuthenticatedURL"
+      },
+      "glob": "log4j.xml"
+    }
+  ]
+}

--- a/metadata/org.apache.hadoop/hadoop-auth/index.json
+++ b/metadata/org.apache.hadoop/hadoop-auth/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.8.2",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/hadoop/hadoop-auth/$version$/hadoop-auth-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/hadoop",
+    "test-code-url" : "https://github.com/apache/hadoop/tree/rel/release-$version$/hadoop-common-project/hadoop-auth/src/test",
+    "documentation-url" : "https://hadoop.apache.org/docs/r$version$/hadoop-auth/index.html",
+    "description" : "Apache Hadoop Auth is a Java HTTP authentication library that provides client and server components for Kerberos SPNEGO authentication over HTTP. It also offers pluggable authentication mechanisms, including pseudo/simple authentication, for Hadoop-compatible HTTP services.",
     "tested-versions" : [
       "2.8.2"
     ],

--- a/metadata/org.apache.hadoop/hadoop-auth/index.json
+++ b/metadata/org.apache.hadoop/hadoop-auth/index.json
@@ -1,6 +1,16 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.8.2",
+    "tested-versions" : [
+      "2.8.2"
+    ],
+    "allowed-packages" : [
+      "org.apache.hadoop.security.authentication",
+      "org.apache.hadoop.util"
+    ]
+  },
+  {
     "metadata-version" : "2.2.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/hadoop/hadoop-auth/$version$/hadoop-auth-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/hadoop",

--- a/stats/org.apache.hadoop/hadoop-auth/2.8.2/execution-metrics.json
+++ b/stats/org.apache.hadoop/hadoop-auth/2.8.2/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_java_run_fail:2026-05-18": {
+    "timestamp": "2026-05-18T12:48:58.321980Z",
+    "library": "org.apache.hadoop:hadoop-auth:2.8.2",
+    "starting_commit": "5ac1bfb193c9ea9b387e726daf95632ceeb30ae9",
+    "ending_commit": "6b64dc6923e20ffdd710152377d15d5b99c3e94f",
+    "previous_library": "org.apache.hadoop:hadoop-auth:2.2.0",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.8.2",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 8,
+            "totalCalls": 8
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 8,
+        "totalCalls": 8
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 383,
+          "missed": 6317,
+          "ratio": 0.057164,
+          "total": 6700
+        },
+        "line": {
+          "covered": 97,
+          "missed": 1449,
+          "ratio": 0.062743,
+          "total": 1546
+        },
+        "method": {
+          "covered": 27,
+          "missed": 241,
+          "ratio": 0.100746,
+          "total": 268
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.2.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 5,
+            "totalCalls": 5
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 5,
+        "totalCalls": 5
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 204,
+          "missed": 3049,
+          "ratio": 0.062711,
+          "total": 3253
+        },
+        "line": {
+          "covered": 55,
+          "missed": 685,
+          "ratio": 0.074324,
+          "total": 740
+        },
+        "method": {
+          "covered": 18,
+          "missed": 114,
+          "ratio": 0.136364,
+          "total": 132
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 72249,
+      "cached_input_tokens_used": 836096,
+      "output_tokens_used": 10765,
+      "iterations": 2,
+      "cost_usd": 0.741,
+      "tested_library_loc": 97,
+      "metadata_entries": 11,
+      "previous_library_metadata_entries": 10,
+      "code_coverage_percent": 6.27,
+      "previous_library_coverage_percent": 7.43
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/KerberosUtilTest.java",
+      "metadata_file": "metadata/org.apache.hadoop/hadoop-auth/2.8.2/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.apache.hadoop/hadoop-auth/2.8.2/stats.json
+++ b/stats/org.apache.hadoop/hadoop-auth/2.8.2/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.8.2",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 8,
+          "totalCalls" : 8
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 8,
+      "totalCalls" : 8
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 383,
+        "missed" : 6317,
+        "ratio" : 0.057164,
+        "total" : 6700
+      },
+      "line" : {
+        "covered" : 97,
+        "missed" : 1449,
+        "ratio" : 0.062743,
+        "total" : 1546
+      },
+      "method" : {
+        "covered" : 27,
+        "missed" : 241,
+        "ratio" : 0.100746,
+        "total" : 268
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/.gitignore
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/build.gradle
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/build.gradle
@@ -1,0 +1,43 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.apache.hadoop:hadoop-auth:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+tasks.withType(Test).configureEach {
+    jvmArgs += [
+        '--add-opens=java.security.jgss/sun.security.jgss=ALL-UNNAMED',
+        '--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED'
+    ]
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+    binaries {
+        all {
+            buildArgs.addAll([
+                '--enable-url-protocols=http',
+                '--add-opens=java.security.jgss/sun.security.jgss=ALL-UNNAMED',
+                '--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED'
+            ])
+        }
+    }
+}

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/gradle.properties
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.apache.hadoop:hadoop-auth:2.8.2
+library.version = 2.8.2
+metadata.dir = org.apache.hadoop/hadoop-auth/2.8.2/

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/settings.gradle
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.apache.hadoop.hadoop-auth_tests'

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java
@@ -1,0 +1,39 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_hadoop.hadoop_auth;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.apache.hadoop.security.authentication.client.AuthenticatedURL;
+import org.apache.hadoop.security.authentication.client.Authenticator;
+import org.apache.hadoop.security.authentication.client.KerberosAuthenticator;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AuthenticatedURLTest {
+    @Test
+    void opensConnectionWithDefaultAuthenticatorWhenTokenIsAlreadySet() throws Exception {
+        Class<? extends Authenticator> previousAuthenticator = AuthenticatedURL.getDefaultAuthenticator();
+        AuthenticatedURL.setDefaultAuthenticator(KerberosAuthenticator.class);
+        try {
+            AuthenticatedURL.Token token = new AuthenticatedURL.Token("delegation-token");
+            AuthenticatedURL authenticatedURL = new AuthenticatedURL();
+
+            HttpURLConnection connection = authenticatedURL.openConnection(new URL("http://localhost/"), token);
+            try {
+                assertThat(connection.getRequestProperty("Cookie"))
+                    .isEqualTo("hadoop.auth=\"delegation-token\"");
+            } finally {
+                connection.disconnect();
+            }
+        } finally {
+            AuthenticatedURL.setDefaultAuthenticator(previousAuthenticator);
+        }
+    }
+}

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/AuthenticatedURLTest.java
@@ -6,9 +6,16 @@
  */
 package org_apache_hadoop.hadoop_auth;
 
+import java.io.IOException;
 import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.atomic.AtomicReference;
 
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpServer;
 import org.apache.hadoop.security.authentication.client.AuthenticatedURL;
 import org.apache.hadoop.security.authentication.client.Authenticator;
 import org.apache.hadoop.security.authentication.client.KerberosAuthenticator;
@@ -22,18 +29,63 @@ public class AuthenticatedURLTest {
         Class<? extends Authenticator> previousAuthenticator = AuthenticatedURL.getDefaultAuthenticator();
         AuthenticatedURL.setDefaultAuthenticator(KerberosAuthenticator.class);
         try {
-            AuthenticatedURL.Token token = new AuthenticatedURL.Token("delegation-token");
-            AuthenticatedURL authenticatedURL = new AuthenticatedURL();
+            AtomicReference<String> cookieHeader = new AtomicReference<>();
+            try (TestHttpServer server = TestHttpServer.create(exchange -> {
+                cookieHeader.set(exchange.getRequestHeaders().getFirst("Cookie"));
+                byte[] response = "ok".getBytes(StandardCharsets.UTF_8);
+                exchange.sendResponseHeaders(HttpURLConnection.HTTP_OK, response.length);
+                exchange.getResponseBody().write(response);
+            })) {
+                AuthenticatedURL.Token token = new AuthenticatedURL.Token("delegation-token");
+                AuthenticatedURL authenticatedURL = new AuthenticatedURL();
 
-            HttpURLConnection connection = authenticatedURL.openConnection(new URL("http://localhost/"), token);
-            try {
-                assertThat(connection.getRequestProperty("Cookie"))
-                    .isEqualTo("hadoop.auth=\"delegation-token\"");
-            } finally {
-                connection.disconnect();
+                HttpURLConnection connection = authenticatedURL.openConnection(server.url(), token);
+                try {
+                    connection.setConnectTimeout(5_000);
+                    connection.setReadTimeout(5_000);
+                    assertThat(connection.getResponseCode()).isEqualTo(HttpURLConnection.HTTP_OK);
+                    assertThat(cookieHeader.get()).isEqualTo("hadoop.auth=\"delegation-token\"");
+                } finally {
+                    connection.disconnect();
+                }
             }
         } finally {
             AuthenticatedURL.setDefaultAuthenticator(previousAuthenticator);
         }
+    }
+
+    private static final class TestHttpServer implements AutoCloseable {
+        private final HttpServer server;
+
+        private TestHttpServer(HttpServer server) {
+            this.server = server;
+        }
+
+        private static TestHttpServer create(ThrowingHttpHandler handler) throws IOException {
+            HttpServer server = HttpServer.create(
+                new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 0);
+            server.createContext("/", exchange -> {
+                try (exchange) {
+                    handler.handle(exchange);
+                }
+            });
+            server.start();
+            return new TestHttpServer(server);
+        }
+
+        private URL url() throws IOException {
+            return new URL("http", server.getAddress().getHostString(),
+                server.getAddress().getPort(), "/");
+        }
+
+        @Override
+        public void close() {
+            server.stop(0);
+        }
+    }
+
+    @FunctionalInterface
+    private interface ThrowingHttpHandler {
+        void handle(HttpExchange exchange) throws IOException;
     }
 }

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/KerberosUtilTest.java
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/KerberosUtilTest.java
@@ -6,6 +6,7 @@
  */
 package org_apache_hadoop.hadoop_auth;
 
+import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
@@ -26,27 +27,57 @@ public class KerberosUtilTest {
 
     @Test
     void readsDefaultRealmFromConfiguredKrb5File(@TempDir Path tempDir) throws Exception {
-        Path krb5Conf = tempDir.resolve("krb5.conf");
-        Files.writeString(krb5Conf, """
-                [libdefaults]
-                    default_realm = EXAMPLE.COM
-
-                [realms]
-                    EXAMPLE.COM = {
-                        kdc = localhost
-                    }
-                """);
+        Path krb5Conf = writeKrb5Conf(tempDir);
 
         String previousKrb5Conf = System.getProperty("java.security.krb5.conf");
         System.setProperty("java.security.krb5.conf", krb5Conf.toString());
         try {
-            assertThat(KerberosUtil.getDefaultRealm()).isEqualTo("EXAMPLE.COM");
+            assertThat(KerberosUtil.getDefaultRealm()).isEqualTo("DEFAULT.EXAMPLE.COM");
         } finally {
-            if (previousKrb5Conf == null) {
-                System.clearProperty("java.security.krb5.conf");
-            } else {
-                System.setProperty("java.security.krb5.conf", previousKrb5Conf);
-            }
+            restoreKrb5Conf(previousKrb5Conf);
+        }
+    }
+
+    @Test
+    void mapsServiceHostDomainToConfiguredRealm(@TempDir Path tempDir) throws Exception {
+        Path krb5Conf = writeKrb5Conf(tempDir);
+
+        String previousKrb5Conf = System.getProperty("java.security.krb5.conf");
+        System.setProperty("java.security.krb5.conf", krb5Conf.toString());
+        try {
+            assertThat(KerberosUtil.getDomainRealm("HTTP/service.example.com"))
+                    .isEqualTo("DOMAIN.EXAMPLE.COM");
+        } finally {
+            restoreKrb5Conf(previousKrb5Conf);
+        }
+    }
+
+    private static Path writeKrb5Conf(Path tempDir) throws IOException {
+        Path krb5Conf = tempDir.resolve("krb5.conf");
+        Files.writeString(krb5Conf, """
+                [libdefaults]
+                    default_realm = DEFAULT.EXAMPLE.COM
+
+                [realms]
+                    DEFAULT.EXAMPLE.COM = {
+                        kdc = localhost
+                    }
+                    DOMAIN.EXAMPLE.COM = {
+                        kdc = localhost
+                    }
+
+                [domain_realm]
+                    example.com = DOMAIN.EXAMPLE.COM
+                    .example.com = DOMAIN.EXAMPLE.COM
+                """);
+        return krb5Conf;
+    }
+
+    private static void restoreKrb5Conf(String previousKrb5Conf) {
+        if (previousKrb5Conf == null) {
+            System.clearProperty("java.security.krb5.conf");
+        } else {
+            System.setProperty("java.security.krb5.conf", previousKrb5Conf);
         }
     }
 }

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/KerberosUtilTest.java
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/src/test/java/org_apache_hadoop/hadoop_auth/KerberosUtilTest.java
@@ -1,0 +1,52 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_hadoop.hadoop_auth;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.hadoop.security.authentication.util.KerberosUtil;
+import org.ietf.jgss.Oid;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KerberosUtilTest {
+    @Test
+    void getsKerberosMechanismOidByName() throws Exception {
+        Oid oid = KerberosUtil.getOidInstance("GSS_KRB5_MECH_OID");
+
+        assertThat(oid.toString()).isEqualTo("1.2.840.113554.1.2.2");
+    }
+
+    @Test
+    void readsDefaultRealmFromConfiguredKrb5File(@TempDir Path tempDir) throws Exception {
+        Path krb5Conf = tempDir.resolve("krb5.conf");
+        Files.writeString(krb5Conf, """
+                [libdefaults]
+                    default_realm = EXAMPLE.COM
+
+                [realms]
+                    EXAMPLE.COM = {
+                        kdc = localhost
+                    }
+                """);
+
+        String previousKrb5Conf = System.getProperty("java.security.krb5.conf");
+        System.setProperty("java.security.krb5.conf", krb5Conf.toString());
+        try {
+            assertThat(KerberosUtil.getDefaultRealm()).isEqualTo("EXAMPLE.COM");
+        } finally {
+            if (previousKrb5Conf == null) {
+                System.clearProperty("java.security.krb5.conf");
+            } else {
+                System.setProperty("java.security.krb5.conf", previousKrb5Conf);
+            }
+        }
+    }
+}

--- a/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/user-code-filter.json
+++ b/tests/src/org.apache.hadoop/hadoop-auth/2.8.2/user-code-filter.json
@@ -1,0 +1,16 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.hadoop.security.authentication.**"
+    },
+    {
+      "includeClasses" : "org.apache.hadoop.util.**"
+    },
+    {
+      "includeClasses" : "org_apache_hadoop.hadoop_auth.**"
+    }
+  ]
+}

